app/src/main/java/com/github/polypoly/app/base/game/Game.kt

```diff
@@ -184,43 +184,55 @@ class Game private constructor(
          */
         fun launchFromPendingGame(gameLobby: GameLobby): Game {
             val players = gameLobby.usersRegistered.map {
-                Player(it, gameLobby.rules.initialPlayerBalance, listOf())
+                Player(it, gameLobby.rules.initialPlayerBalance)
+            }
+            val inGameLocations = gameLobby.rules.gameMap.flatMap { zone ->
+                zone.locationProperties.map { InGameLocation(it) }
             }
+            if(gameLobby.rules.gameMode == GameMode.LANDLORD)
+                assignRandomLocations(inGameLocations, gameLobby, players)
+
             val game = Game(
                 code = gameLobby.code,
                 admin = gameLobby.admin,
-                players = gameLobby.usersRegistered.map { Player(
-                    user = it,
-                    balance = gameLobby.rules.initialPlayerBalance,
-                    ownedLocations = mutableListOf(),
-                ) },
+                players = players,
                 rules = gameLobby.rules,
                 dateBegin = System.currentTimeMillis() / 1000,
+                inGameLocations = inGameLocations
             )
+
+            GameRepository.player = players.first { it.user.id == (currentUser?.id ?: "") }
```

Comment:
This should be moved back to gameViewModel for better encapsulation

Reply:
I saw you moved it to GameLobby in your last commit, it's fine as well !

Reply:
I move it but not in gameViewModel, I move it in GameLobby.start(), that makes sence here